simulation-construction-set-examples/simple-pendulum-with-controller/src/main/java/us/ihmc/simplePendulum/SimplePendulumDefinition.java

```diff
@@ -0,0 +1,146 @@
+import us.ihmc.scs2.definition.robot.OneDoFJointDefinition;
+import us.ihmc.scs2.definition.robot.RevoluteJointDefinition;
+import us.ihmc.scs2.definition.robot.RigidBodyDefinition;
+import us.ihmc.scs2.definition.robot.RobotDefinition;
+import us.ihmc.scs2.definition.visual.ColorDefinitions;
+import us.ihmc.scs2.definition.visual.MaterialDefinition;
+import us.ihmc.scs2.definition.visual.VisualDefinition;
+
+public class SimplePendulumDefinition extends RobotDefinition
+{
+   private static final String PENDULUM = "pendulum";
+
+   // Define the parameters of the robot 
+   public static final double ROD_LENGTH = 1.0;
+   public static final double ROD_RADIUS = 0.01;
+   public static final double ROD_MASS = 0.00;
+   public static final double TIP_RADIUS = 0.02;
+
+   public static final double FULCRUM_RADIUS = 0.02;
+
+   public static final double BALL_RADIUS = 0.05;
+   public static final double BALL_MASS = 1.0;
+
+   // I = mr^2 pendulum's resistance changes to its rotation in kg.m^2
+   public static final double FULCRUM_MOMENT_OF_INERTIA_ABOUT_Y = BALL_MASS * ROD_LENGTH * ROD_LENGTH;
+
+   // Initial state of the pendulum
+   private double fulcrumInitialPositionDegrees = 45.0;
+   private double fulcrumInitialPositionRadians = fulcrumInitialPositionDegrees * Math.PI / 180.0;
+   private double fulcrumInitialVelocity = 0.0;
+
+   private static final double DAMP = 0.06;
+
+   private final OneDoFJointDampingControllerDefinition jointDampingControllerDefinition = new OneDoFJointDampingControllerDefinition();
+
+   /*
+    * private final OneDoFJointDampingControllerDefinition jointLvl1DampingControllerDefinition = new
+    * OneDoFJointDampingControllerDefinition(); Some joint state variables. Allows SimplePendulumRobot
+    * to have access to and set fulcrum joint properties.
+    */
+
+   // Define its constructor
+   public SimplePendulumDefinition()
+   {
+      // Call parent class "RobotDefinition" constructor. The string PENDULUM will be the name of the robot.
+      super(PENDULUM);
+      // creates an instance of the class RobotDefinition named "pendulum" in the SCS2 system.
+
+      // Create the top (fixed) link that serves as the base of the pendulum
+      RigidBodyDefinition elevator = new RigidBodyDefinition("elevator");
+      setRootBodyDefinition(elevator);
+
+      /*
+       * The first parameter "FulcrumPin" is the name of the joint and will be used in all the variables
+       * associated with the joint. The second parameter "new Vector3d(0.0, 0.0, 1.5)" defines the offset
+       * of this joint from the previous joint. Since we want to position the fulcrum of the pendulum at a
+       * height of 1.5 meters above the ground, the default vector (0.0, 0.0, 1.5) will be used. The
+       * parameter "Axis3D.Y" defines the axis of rotation for this pin joint.
+       */
+      OneDoFJointDefinition fulcrumPinJoint = new RevoluteJointDefinition("FulcrumPin", new Vector3D(0.0, 0.0, 1.5), Axis3D.Y);
```

Comment:
Let's make this joint name a public static String that the controller can use to look up for the joint

Reply:
Done.